src/Microsoft.Tye.Core/KubernetesManifestGenerator.cs

```diff
@@ -452,6 +462,64 @@ internal static class KubernetesManifestGenerator
             return new KubernetesDeploymentOutput(project.Name, new YamlDocument(root));
         }
 
+        private static void AddProbe(ServiceBuilder service, YamlMappingNode container, ProbeBuilder builder, string name)
+        {
+            var probe = new YamlMappingNode();
+            container.Add(name, probe);
+
+            if (builder.Http != null)
+            {
+                var builderHttp = builder.Http!;
+                var httpGet = new YamlMappingNode();
+
+                probe.Add("httpGet", httpGet);
+                httpGet.Add("path", builderHttp.Path);
+
+                if (builderHttp.Protocol != null)
+                {
+                    httpGet.Add("scheme", builderHttp.Protocol!.ToUpper());
```

Comment:
I think you don't need `!` here, other similar cases in this file as well.

```suggestion
                    httpGet.Add("scheme", builderHttp.Protocol.ToUpper());
```

Reply:
yep. I removed it and it didn't throw any warnings. probably because it's inside a block that runs if `builderHttp.Protocol != null`.
It's weird because at times the compiler will throw a warning when not using the `!` operator, even though the expression is in a context where it's guaranteed to not be null...

Reply:
found an example: if you go to `ReplicaMonitor.cs`:113 and remove the `!` after `serviceDesc.Readiness`, the compiler will throw a warning, even though there is no way for that variable to be null at that point.